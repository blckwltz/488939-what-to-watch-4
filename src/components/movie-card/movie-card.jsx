import React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import {Link} from 'react-router-dom';
import {AppRoute} from '../../routing/route.js';
import {getFeaturedMovie} from '../../store/movies/selectors.js';
import withStatus from '../../hocs/with-status/with-status.jsx';
import UserBlock from '../user-block/user-block.jsx';
import FavoriteButton from '../favorite-button/favorite-button.jsx';

const FavoriteButtonWrapped = withStatus(FavoriteButton);

const MovieCard = (props) => {
  const {movie} = props;
  const {id, isFavorite, title, genre, releaseDate, cover, backgroundImage, backgroundColor} = movie;

  return <section className="movie-card" style={{background: backgroundColor}}>
    <div className="movie-card__bg">
      <img src={backgroundImage} alt={title}/>
    </div>

    <h1 className="visually-hidden">WTW</h1>

    <header className="page-header movie-card__head">
      <div className="logo">
        <a className="logo__link">
          <span className="logo__letter logo__letter--1">W</span>
          <span className="logo__letter logo__letter--2">T</span>
          <span className="logo__letter logo__letter--3">W</span>
        </a>
      </div>

      <UserBlock/>
    </header>

    <div className="movie-card__wrap">
      <div className="movie-card__info">
        <div className="movie-card__poster">
          <img src={cover} alt={title} width="218"
            height="327"/>
        </div>

        <div className="movie-card__desc">
          <h2 className="movie-card__title">{title}</h2>
          <p className="movie-card__meta">
            <span className="movie-card__genre">{genre}</span>
            <span className="movie-card__year">{releaseDate}</span>
          </p>

          <div className="movie-card__buttons">
            <Link to={`${AppRoute.PLAYER}/${id}`} className="btn btn--play movie-card__button">
              <svg viewBox="0 0 19 19" width="19" height="19">
                <use xlinkHref="#play-s"/>
              </svg>
              <span>Play</span>
            </Link>
            <FavoriteButtonWrapped id={id} isFavorite={isFavorite}/>
          </div>
        </div>
      </div>
    </div>
  </section>;
};

MovieCard.propTypes = {
  movie: PropTypes.shape({
    id: PropTypes.number,
    isFavorite: PropTypes.bool,
    title: PropTypes.string,
    genre: PropTypes.string,
    releaseDate: PropTypes.number,
    cover: PropTypes.string,
    backgroundImage: PropTypes.string,
    backgroundColor: PropTypes.string,
  }).isRequired,
};

const mapStateToProps = (state) => ({
  movie: getFeaturedMovie(state),
});

export {MovieCard};
export default connect(mapStateToProps)(MovieCard);
