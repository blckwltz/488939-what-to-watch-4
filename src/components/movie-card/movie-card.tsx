import React from 'react';
import {connect} from 'react-redux';
import {Link} from 'react-router-dom';
import {Movie} from '../../types/movie';
import {AppRoute} from '../../routing/route';
import {getFeaturedMovie} from '../../store/movies/selectors';
import withStatus from '../../hocs/with-status/with-status';
import UserBlock from '../user-block/user-block';
import FavoriteButton from '../favorite-button/favorite-button';

interface Props {
  movie: Movie,
}

const FavoriteButtonWrapped = withStatus(FavoriteButton);

const MovieCard = (props: Props) => {
  const {movie} = props;
  const {id, isFavorite, title, genre, releaseDate, cover, backgroundImage, backgroundColor} = movie;

  return <section className="movie-card" style={{background: backgroundColor}}>
    <div className="movie-card__bg">
      <img src={backgroundImage} alt={title}/>
    </div>

    <h1 className="visually-hidden">WTW</h1>

    <header className="page-header movie-card__head">
      <div className="logo">
        <a className="logo__link">
          <span className="logo__letter logo__letter--1">W</span>
          <span className="logo__letter logo__letter--2">T</span>
          <span className="logo__letter logo__letter--3">W</span>
        </a>
      </div>

      <UserBlock/>
    </header>

    <div className="movie-card__wrap">
      <div className="movie-card__info">
        <div className="movie-card__poster">
          <img src={cover} alt={title} width="218"
            height="327"/>
        </div>

        <div className="movie-card__desc">
          <h2 className="movie-card__title">{title}</h2>
          <p className="movie-card__meta">
            <span className="movie-card__genre">{genre}</span>
            <span className="movie-card__year">{releaseDate}</span>
          </p>

          <div className="movie-card__buttons">
            <Link to={`${AppRoute.PLAYER}/${id}`} className="btn btn--play movie-card__button">
              <svg viewBox="0 0 19 19" width="19" height="19">
                <use xlinkHref="#play-s"/>
              </svg>
              <span>Play</span>
            </Link>
            <FavoriteButtonWrapped id={id} isFavorite={isFavorite}/>
          </div>
        </div>
      </div>
    </div>
  </section>;
};

const mapStateToProps = (state) => ({
  movie: getFeaturedMovie(state),
});

export {MovieCard};
export default connect(mapStateToProps)(MovieCard);
