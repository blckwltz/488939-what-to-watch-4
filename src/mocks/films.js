export default {
  featured: {
    title: `The Grand Budapest Hotel`,
    genre: `Drama`,
    releaseDate: 2014,
  },
  list: [
    {
      title: `Macbeth`,
      poster: `img/macbeth.jpg`,
    },
    {
      title: `We need to talk about Kevin`,
      poster: `img/we-need-to-talk-about-kevin.jpg`,
    },
    {
      title: `Fantastic Beasts: The Crimes of Grindelwald`,
      poster: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    },
    {
      title: `Shutter Island`,
      poster: `img/shutter-island.jpg`,
    },
    {
      title: `Snatch`,
      poster: `img/snatch.jpg`,
    },
    {
      title: `Pulp Fiction`,
      poster: `img/pulp-fiction.jpg`,
    },
    {
      title: `Revenant`,
      poster: `img/revenant.jpg`,
    },
    {
      title: `Bohemian Rhapsody`,
      poster: `img/bohemian-rhapsody.jpg`,
    },
  ]
};
