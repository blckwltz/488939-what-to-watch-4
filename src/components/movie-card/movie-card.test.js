import React from 'react';
import renderer from 'react-test-renderer';
import {featuredMovie} from '../../mocks/films';
import {MovieCard} from './movie-card';

it(`Should render MovieCard component correctly`, () => {
  const tree = renderer
    .create(
        <MovieCard
          movie={featuredMovie}
        />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
