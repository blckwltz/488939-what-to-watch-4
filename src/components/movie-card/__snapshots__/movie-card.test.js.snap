// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render correctly 1`] = `
<article
  className="small-movie-card catalog__movies-card"
>
  <div
    className="small-movie-card__image"
  >
    <img
      alt="The Big Lebowski"
      height="175"
      src="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
      width="280"
    />
  </div>
  <h3
    className="small-movie-card__title"
  >
    <a
      className="small-movie-card__link"
      href="movie-page.html"
      onClick={[Function]}
    >
      The Big Lebowski
    </a>
  </h3>
</article>
`;
